### PR TITLE
chia: fix farming with a pool 

### LINKED ISSUE #131809: chia: cannot farm with a pool
### Describe the bug
chia farmer fails to register with a pool over https. Relevant log contains:
```
2021-07-28T14:09:59.519 farmer chia.farmer.farmer         : ERROR    Exception in GET /pool_info https://«pool», [Errno 2] No such file or directory
2021-07-28T14:09:59.520 farmer chia.farmer.farmer         : WARNING  No pool specific authentication_token_timeout has been set for f364bb759feda74651c60b82c34bf6cdf8c14e9d642c583b7dd0f0d66fbee5ae, check communication with the pool
```

### Steps To Reproduce
1. have a new-style chia plot
2. register with a pool
3. set up a farming node

### Expected behavior
Chia should successfully run.

### Additional context
Based on `strace`, the missing file is `/nix/store/qp7dnwy19x9qnvfhv00ai668fviwwp8a-chia-1.2.3/lib/python3.9/site-packages/mozilla-ca/cacert.pem`. Upstream normally [bundles mozilla-ca](https://github.com/Chia-Network/chia-blockchain/blob/main/setup.py#L109) but this appears to fail here.

### Notify maintainers
@Atemu @lourkeur 

### Metadata
(This is in an LXC container so it's a bit weird)
 - system: `0`
 - host os: `Linux 5.10.17, NixOS, 21.05.20210726.91903ce (Okapi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210707_02dd6bb`
 - nixpkgs: `/nix/store/7xmrq7y1irmnadkwy3dsw0asf6zrmsxk-source`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - chia
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
#131809

###### Things done
Fetch submodules. This appears to be sufficient, my production setup is fixed.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — CHANGES_REQUESTED** [review]
please change the commit message to:

```
chia: fix farming with a pool by fetching submodules

https://github.com/NixOS/nixpkgs/issues/131809
```

File: pkgs/applications/blockchains/chia/default.nix [hunk 1]
```diff
@@ -12,7 +12,8 @@ python3Packages.buildPythonApplication rec {
     owner = "Chia-Network";
     repo = "chia-blockchain";
     rev = version;
-    sha256 = "sha256-yS0/Fy2dj8VIbwv2J9sehP0VN0f/YDxu1k9WkaeEz8M=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/chia/default.nix:15]: I am fine with vendoring some typescript/js https://github.com/Chia-Network/chia-blockchain-gui but vendoring a ca is not fine for me https://github.com/Chia-Network/mozilla-ca. We should use the nix one.

Comment (lourkeur) [reply pkgs/applications/blockchains/chia/default.nix:15]: > I am fine with vendoring some typescript/js https://github.com/Chia-Network/chia-blockchain-gui but vendoring a ca is not fine for me https://github.com/Chia-Network/mozilla-ca. We should use the nix one.

I am confused as to what derivation the canonical Mozilla CA bundle lives in. please tell me.

[`nix-locate cacert.pem`](https://gist.github.com/lourkeur/bd389ba7922bb952b342cbaf55992c87)

Comment (Artturin) [reply pkgs/applications/blockchains/chia/default.nix:15]: just use `cacert` i think

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/applications/blockchains/chia/default.nix (hunk 1 above)
Comment (ilyakooo0) [reply pkgs/applications/blockchains/chia/default.nix:16]: Do we still need the submodules if we are using `cacert`?

Comment (lourkeur) [reply pkgs/applications/blockchains/chia/default.nix:16]: good question: yes we do, there are still one useful file in that module besides the CA bundle. (currently an empty `__init__.py`, but that could change) Besides, there's also the gui bits which may become useful later.

**Pull request merged**